src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -171,23 +161,18 @@ public DynamicObject readStringToNull(DynamicObject pointer,
             final byte[] bytes;
 
             if (TruffleOptions.AOT) {
-                final jnr.ffi.Pointer ptr = Layouts.POINTER.getPointer(pointer);
-                final int nullOffset = findNullOffset(ptr);
+                final Pointer ptr = Layouts.POINTER.getPointer(pointer);
+                final int nullOffset = ptr.findByte(0, (byte) 0);
                 bytes = new byte[nullOffset];
 
-                PointerOperations.readPointer(ptr, bytes, nullOffset);
+                ptr.readBytes(0, bytes, 0, nullOffset);
```

Comment:
We can use `Pointer#readZeroTerminatedByteArray` here, and for both branches (AOT and not).